src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.html

```diff
@@ -0,0 +1,343 @@
+        />
+      </cds-text-label>
+      <ng-template #clusterError>
+        <span
+          class="invalid-feedback"
+          *ngIf="smbForm.showError('cluster_id', formDir, 'required')"
+          i18n
+          >This field is required.</span
+        >
+      </ng-template>
+    </div>
+
+    <!-- Auth Mode -->
+    <div class="form-item">
+      <cds-select
+        formControlName="auth_mode"
+        name="auth_mode"
+        for="auth_mode"
+        label="Auth Mode"
+        cdRequiredField="Auth Mode"
+        id="auth_mode"
+        [invalid]="smbForm.controls.auth_mode.invalid && smbForm.controls.auth_mode.dirty"
+        [invalidText]="authModeError"
+        (change)="onAuthModeChange()"
+        i18n
+      >
+        <option value="">-- Select the auth mode --</option>
+        <option value="active-directory">Active Directory</option>
+        <option value="user">User</option>
+      </cds-select>
+      <cd-help-text *ngIf="this.smbForm.get('auth_mode').value === 'active-directory'">
+        Active directory manages all user information and user authentication for the cluster.
+      </cd-help-text>
+      <cd-help-text *ngIf="this.smbForm.get('auth_mode').value === 'user'">
+        The user authenticates and manages their own credentials within the cluster.
+      </cd-help-text>
+      <ng-template #authModeError>
+        <span
+          class="invalid-feedback"
+          *ngIf="smbForm.showError('auth_mode', formDir, 'required')"
+          i18n
+          >This field is required.</span
+        >
+      </ng-template>
+    </div>
+
+    <!-- Domain Settings -->
+    <div class="form-item"
+         *ngIf="this.smbForm.get('auth_mode').value === 'active-directory'">
+      <div cdsCol
+           [columnNumbers]="{ md: 12 }"
+           class="d-flex">
+        <cds-text-label for="domain_settings"
+                        i18n
+                        cdRequiredField="Domain Settings"
+          >Domain Settings
+          <div class="cds-input-group">
+            <input
+              cdsText
+              type="text"
+              placeholder="Domain Settings..."
+              i18n-placeholder
+              id="domain_settings"
+              name="domain_settings"
+              formControlName="domain_settings"
+              [value]="domainSettingsObject?.realm"
+              (click)="editDomainSettingsModal()"
+              [invalid]="
+                !smbForm.controls.domain_settings.valid && smbForm.controls.domain_settings.dirty
+              "
+            />
+            <cds-icon-button kind="ghost"
+                             (click)="editDomainSettingsModal()"
+                             size="md">
+              <svg cdsIcon="edit"
+                   size="32"
+                   class="cds--btn__icon"
+                  icon></svg>
+            </cds-icon-button>
+
+            <cds-icon-button kind="ghost"
```

Comment:
Let's use danger for delete buttons
```suggestion
            <cds-icon-button kind="danger"
```